mqtt/mqttd/src/broker/bootstrap/edgehub.rs

```diff
@@ -165,50 +165,35 @@ async fn server_certificate_renewal(renew_at: DateTime<Utc>) {
 async fn start_sidecars(
     broker_handle: BrokerHandle,
     system_address: String,
-) -> (SidecarShutdownHandle, JoinHandle<Result<()>>) {
-    let (termination_handle, tx) = channel::<()>();
+) -> Result<(SidecarShutdownHandle, JoinHandle<Result<()>>)> {
+    let (sidecar_termination_handle, sidecar_termination_receiver) = channel::<()>();
 
-    let event_loop = tokio::spawn(async move {
-        info!("starting command handler...");
-        let (mut command_handler_shutdown_handle, command_handler_join_handle) =
-            start_command_handler(broker_handle, system_address).await?;
+    let device_id = env::var(DEVICE_ID_ENV)?;
+    let command_handler =
+        CommandHandler::new(broker_handle, system_address, device_id.as_str()).await?;
+    let command_handler_shutdown_handle = command_handler.shutdown_handle()?;
 
-        start_bridge().await?;
+    let mut bridge_controller = BridgeController::new();
```

Comment:
These are fail-able init steps that will percolate errors all the way up. Once the components are initialized we can trust that they will handle errors in the event loop below.